client/src/pages/DirectoryPage/DirectoryPeople.js

```diff
@@ -16,173 +16,324 @@ import image4 from "../../assets/images/profile-picture-4.png";
-          />{" "}
-          <Pill
-            className="Pill"
-            text={"Traveling"}
-            size={"wide"}
-            color={"pink"}
-            icon={"white"}
-            onClick={pillClick}
-            selectId={0}
-          />{" "}
-          <Pill
-            className="Pill"
-            text={"Traveling"}
-            size={"wide"}
-            color={"pink"}
-            icon={"white"}
-            onClick={pillClick}
-            selectId={0}
-          />
-          <Pill
-            className="Pill"
-            text={"Traveling"}
-            size={"wide"}
-            color={"pink"}
-            icon={"white"}
-            onClick={pillClick}
-            selectId={0}
-          />
-          <Pill
-            className="Pill"
-            text={"Traveling"}
-            size={"wide"}
-            color={"pink"}
-            icon={"white"}
-            onClick={pillClick}
-            selectId={0}
-          />
-          <Pill
-            className="Pill"
-            text={"Traveling"}
-            size={"wide"}
-            color={"pink"}
-            icon={"white"}
-            onClick={pillClick}
-            selectId={0}
+      )
+    }
+  }
+
+  showOthers = () => {
+    let cards = []
+    this.otherCards(cards)
+    return cards
+  }
+
+  otherCards = (arr) => {
+    let length  = others.length
+    for (let i = 0; i < length; i++) {
+      //same as the friends function
+      arr.push(
+        <div className="CommunityPage__momCard-single">
+        <PersonCard image={this.state.others[i].image} name={this.state.others[i].first_name} location="Washington D.C." />
+        <button className="FriendButton" onClick={this.sendRequest}>Add Friend</button>
+        </div>
+      )
+    }
+  }
+
+  showRequests = () => {
+    let cards = []
+    this.requestCards(cards)
+    return cards
+  }
+
+  requestCards = (arr) => {
+    let length  = friendRequests.length
+    for (let i = 0; i < length; i++) {
+      //same as the friends function
+      arr.push(
+        <div className="CommunityPage__momCard-single">
+        <PersonCard image={this.state.friendRequests[i].image} name={this.state.friendRequests[i].first_name} location="Washington D.C." />
+        <button className="FriendButton" onClick={this.acceptRequest}>Accept</button>
+        </div>
+      )
+    }
+  }
+
+  trueFriends = (friendId) => {
+    for (let i = 0; i < this.props.connections.length; i++) {
+      if (this.props.connections[i] === friendId) {
+        return true
+      }
+    }
+  }
+
+  removeFriend = (e) => {
+    let nonFriend = e.target.parentElement.children[0].children[1].textContent
+    let connectionsList = this.props.connections
+    for (let i = 0; i < this.state.friends.length; i++) {
+      if (this.state.friends[i].first_name === nonFriend) {
+        connectionsList.splice(i, 1)
+      }
+    }
+    let data = { connections: connectionsList }
+    fetch(`http://127.0.0.1:8000/profile/${this.props.userid}`, {
```

Comment:
Can you put this in a service and use the `base_url` constant? Otherwise, this will break in production